code/game/atom/atom_act.dm

```diff
@@ -201,10 +201,18 @@
  * Causes effects when the atom gets hit by a rust effect from heretics
  *
  * Override this if you want custom behaviour in whatever gets hit by the rust
+ * /turf/rust_turf should be used instead for overriding rust on turfs
  */
 /atom/proc/rust_heretic_act()
 	return
 
+/mob/living/proc/do_rust_heretic_act(atom/target)
+	var/datum/antagonist/heretic/heretic_data = IS_HERETIC(src)
+	target.rust_heretic_act(heretic_data?.rust_strength)
+
+/mob/living/rust_walker/do_rust_heretic_act(atom/target)
```

Comment:
Is this the right typepath? I would think it's `/mob/living/simple_animal/rust_walker` or `/mob/living/basic/rust_walker`